tests/test_windows.py

```diff
@@ -0,0 +1,175 @@
+import rasterio
+
+import numpy as np
+import pytest
+
+from rasterio.windows import (
+    window, window_bounds, window_transform, eval_window, window_index, window_shape)
+
+def test_window_method():
+    with rasterio.open('tests/data/RGB.byte.tif') as src:
+        left, bottom, right, top = src.bounds
+        dx, dy = src.res
+        eps = 1.0e-8
```

Comment:
@perrygeo: semi-random thought here... this could be a great place to try Hypothesis, vary `eps` to make sure our algorithm doesn't burst into flames.